### DISCUSSION TITLE
Aggregation of hierarchical packages

### DISCUSSION THREAD

**Discussion by nyarly** [cue-lang/cue#1108] [General]
I was wondering about the way that the `cue` tool merges parent packages with their children. 
I can't find a direct reference to it any more; I remember that 
https://github.com/cue-lang/cue/blob/v0.4.0/doc/tutorial/kubernetes/README.md
makes use of the concept.
So, I'm not sure how exactly to call the feature.
Is that a facility that folks make general use of? It seemed like a neat idea, but I wound up needing to use Kustomize with my Kubernetes configuration, so the file structure layout of Kustomize is in tension with the "stacking" features.
Conversely, the fact of the stacking behavior pushes my CUE source into a flat hierarchy,
which at least rankles my aesthetic sense.

Have I wandered out into the wilderness in this respect? I know others have built `cue` based Go programs, which presumably wouldn't use this feature either.

**Comment by myitcv** [comment]
Thanks for raising this @nyarly. Generally I think you are referring to this section of the docs:

https://cuelang.org/docs/concepts/packages/#file-organization

(which is outside of the language, and a function of the `cmd/cue` tool).

This particular model instance specialisation is one approach to file organisation. As I mentioned on our call, https://github.com/timberio/vector has taken a very different approach, where they have multiple package files split across subdirectories and then do the equivalent of `cue eval **/*.cue` (using shell globbing) to evaluate as a single instance. 

We would be very interested in hearing experience reports/ideas from folks on whether we should expand `cmd/cue` to support other models that better align with different use cases.

Please can you provide some more details about the Kustomize model? What sort of layout and loading would work for you?

**Comment by verdverm** [comment]
Related: https://github.com/cue-lang/cue/discussions/411
